docs/t-sql/functions/count-transact-sql.md

```diff
@@ -58,20 +58,26 @@ OVER **(** [ *partition_by_clause* ] [ *order_by_clause* ] [ *ROW_or_RANGE_claus
 
 ## Return types
- **int**  
+ 
+ * **int NOT NULL** when `ANSI_WARNINGS` is `ON`, however SQL Server will always treat `COUNT` expressions as `int NULL` in metadata, unless wrapped in `ISNULL`.
+ * **int NULL** when `ANSI_WARNINGS` is `OFF`.
   
 ## Remarks  
-COUNT(\*) returns the number of items in a group. This includes NULL values and duplicates.
-  
-COUNT(ALL *expression*) evaluates *expression* for each row in a group, and returns the number of nonnull values.
-  
-COUNT(DISTINCT *expression*) evaluates *expression* for each row in a group, and returns the number of unique, nonnull values.
-  
-For return values exceeding 2^31-1, `COUNT` returns an error. For these cases, use `COUNT_BIG` instead.
+
+* `COUNT(\*)` without `GROUP BY` returns the cardinality (number of rows) in the resultset. This includes rows comprised of all-`NULL` values and duplicates.
+* `COUNT(\*)` with `GROUP BY` returns the number of rows in each group. This includes `NULL` values and duplicates.
+* `COUNT(ALL *expression*)` evaluates *expression* for each row in a group, and returns the number of nonnull values.
+* `COUNT(DISTINCT *expression*)` evaluates *expression* for each row in a group, and returns the number of unique, nonnull values.
   
 `COUNT` is a deterministic function when used ***without*** the OVER and ORDER BY clauses. It is nondeterministic when used ***with*** the OVER and ORDER BY clauses. See [Deterministic and Nondeterministic Functions](../../relational-databases/user-defined-functions/deterministic-and-nondeterministic-functions.md) for more information.
+
+### `ARITHABORT` and `ANSI_WARNINGS`
+
+When `COUNT` will have a return value exceeding the maximum value of `int` (i.e. 2<sup>31</sup>-1 or 2,147,483,647) then the `COUNT` function will fail due to integer overflow. When `COUNT` overflows and _both_ the `ARITHABORT` and `ANSI_WARNINGS` options are `OFF` then `COUNT` will return `NULL`; otherwise (when _either_ of `ARITHABORT` or `ANSI_WARNINGS` are `ON`) then the query will be aborted and the arithmetic overflow error "Msg 8115, Level 16, State 2; Arithmetic overflow error converting expression to data type int." will be raised. To correctly handle these large results instead use `COUNT_BIG` which returns `bigint`.
+
+When both ``ARITHABORT`` and `ANSI_WARNINGS` are `ON` you can safely wrap `COUNT` call-sites in `ISNULL( <count-expr>, 0 )` to coerce the expression's type to `int NOT NULL` instead of `int NULL`; otherwise then wrapping `COUNT` is `ISNULL` means any overflow error will be silently suppressed which should be carefully considered for correctness.

```

Comment:
```suggestion
When both ``ARITHABORT`` and `ANSI_WARNINGS` are `ON`, you can safely wrap `COUNT` call-sites in `ISNULL( <count-expr>, 0 )` to coerce the expression's type to `int NOT NULL` instead of `int NULL`. Wrapping `COUNT` in `ISNULL` means any overflow error will be silently suppressed, which should be considered for correctness.
```